src/Processors/Formats/Impl/ArrowBufferedStreams.cpp

```diff
@@ -132,6 +135,12 @@ std::shared_ptr<arrow::io::RandomAccessFile> asArrowFile(ReadBuffer & in)
             return std::make_shared<RandomAccessFileFromSeekableReadBuffer>(*fd_in, stat.st_size);
     }
 
+    if (auto * fd_in = dynamic_cast<RemoteReadBuffer *>(&in))
+    {
+        if (fd_in->seekable())
+            return std::make_shared<RandomAccessFileFromSeekableReadBuffer>(*fd_in, fd_in->size());
```

Comment:
This part probably should be reworked after https://github.com/ClickHouse/ClickHouse/pull/30936 is merged.
But not sure for now.

Reply:
I see.

Reply:
It is merged. Please take a look if it can be reused.

Reply:
ok, I will change soon

Reply:
#30936 is a great PR!  We already reused it. Thanks.